src/app.py

```diff
@@ -100,17 +103,43 @@ async def send_scene():
 
     await asyncio.gather(*future_sends)
 
+
 def close_processes():
     for process_data in state.processes.values():
         process_data.process.terminate()
 
+
+async def tick_trigger():
+    if time.time() - state.last_tick >= 1:
+        await tick()
+        state.last_tick = time.time()
+
+async def wait_for_next_tick():
+    tick = state.tick_number
+    while state.tick_number != tick + 1:
+        await asyncio.sleep(0.1)
+
+async def spawn_process_for_new_agent(object: Object):
+    await wait_for_next_tick()
+
+    parent_conn, child_conn = AioPipe()
+    p = AioProcess(target=execution_runtime,
+                    args=(child_conn, object.id,))
+    p.start()
+
+    await parent_conn.coro_send(state.scene.serialize())
+    state.processes[object.id] = ProcessData(
+        process=p, pipe=parent_conn)
```

Comment:
Should be inline with `state.processes[object.id] = ProcessData(`
I think you should set longer line breaks in IDE